### PR TITLE
normalize get available EPs behavior

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
Only get available EP in local and python environment. For AzureML and docker we need let user specify them

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/engine/engine.py [hunk 1]
```diff
@@ -85,19 +85,21 @@ def __init__(
...
             execution_providers = self._config.execution_providers
 
-        # verify the AzureML system have specified the execution providers
-        # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
-        if self.target.system_type == SystemType.AzureML and execution_providers is None:
-            raise ValueError("AzureMLSystem requires execution providers to be specified.")
-        elif execution_providers is None:
-            if self.target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
+        if not execution_providers:
+            if self.target.olive_managed_env:
```
Comment (xiaoyu-work) [inline olive/engine/engine.py:114]: I had a same question in Jambay's PR: is another main idea of `olive_managed_env` that user doesn't need to worry about ep and we will try all possible eps for them? I think that was why we skip `get_supported_execution_providers` call for `olive_managed_env`. Or this is a part of Olive auto optimization?

Comment (guotuofeng) [reply olive/engine/engine.py:114]: yes, for `olive_managed_env` we will create intermediate system and install all the needed ORT.

Comment (xiaoyu-work) [reply olive/engine/engine.py:114]: I mean in this case, should we try all possible eps by device? Does user need to provide eps he wants?

Comment (guotuofeng) [reply olive/engine/engine.py:114]: user need specify what's the EP he/she want to run against multiple EP. and there is no need to try all possible EP

File: olive/engine/engine.py [hunk 2]
```diff
@@ -82,22 +82,49 @@ def __init__(
...
+                # verify the AzureML system have specified the execution providers
+                # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
+                raise ValueError("AzureMLSystem requires execution providers to be specified.")
+            elif self.target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
                 execution_providers = self.target.get_supported_execution_providers()
-            else:
-                # for docker system and python system, we default use CPUExecutionProvider
+            elif self.target.system_type == SystemType.Docker:
+                # for docker system we default use CPUExecutionProvider
                 execution_providers = ["CPUExecutionProvider"]
```
Comment (trajepl) [inline olive/engine/engine.py:124]: From my understanding, in this PR, you want to put off the logic to get EPs if the current system's ORT version unknow(like aml, docker).
Local: SystemType.Local, SystemType.PythonEnvironment
Remote: AML, Docker

Then if so, why we give the CPU eps for docker as default? Should we let user specify for docer system?

File: olive/engine/engine.py [hunk 3]
```diff
@@ -114,17 +141,24 @@ def __init__(
...
-                and self.target.olive_managed_env
-            ):
-                skip_get_available_ep = True
-            supported_eps = AcceleratorLookup.get_execution_providers_for_device(device, skip_get_available_ep)
+            if self.target.olive_managed_env:
+                available_eps = AcceleratorLookup.get_managed_supported_execution_providers(device)
+            elif self.target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
+                available_eps = self.target.get_supported_execution_providers()
+            elif self.target.system_type == SystemType.Docker:
+                # TODO: do we need allow docker system support other execution providers?
```
Comment (trajepl) [inline olive/engine/engine.py:154]: Yes, the TODO item is what I am thinking. See the above comment.

**Review by trajepl — APPROVED** [review]

**Pull request merged**